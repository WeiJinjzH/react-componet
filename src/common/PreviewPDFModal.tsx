import { Modal } from 'antd'
import React, { Component } from 'react'

interface PreviewPDFModalProps {
    params: {
        href: string;
    };
    onDestroy: () => void;
}

interface PreviewPDFModalState {
    visible: boolean;
}

class PreviewPDFModal extends Component<PreviewPDFModalProps, PreviewPDFModalState> {
    constructor(props) {
        super(props)
        this.state = {
            visible: true,
        }
    }

    render() {
        const { href } = this.props.params
        return (
            <Modal
                visible={this.state.visible}
                destroyOnClose
                bodyStyle={{ width: '100%', height: '80vh' }}
                onCancel={() => { this.setState({ visible: false }) }}
                afterClose={this.props.onDestroy}
                footer={null}
            >
                <iframe
                    title="previewer"
                    style={{ width: '100%', height: '100%' }}
                    src={href}
                />
            </Modal>
        )
    }
}

export default PreviewPDFModal
