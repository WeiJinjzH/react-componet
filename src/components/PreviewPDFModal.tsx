import { Modal } from 'antd'
import React, { Component } from 'react'

interface PreviewPDFModalProps {
    params: { href: string; };
    onDestroy: () => void;
}

interface PreviewPDFModalState {
    visible: boolean;
}

class PreviewPDFModal extends Component<PreviewPDFModalProps, PreviewPDFModalState> {
    constructor(props: PreviewPDFModalProps) {
        super(props)
        this.state = {
            visible: true,
        }
    }

    render() {
        const { href } = this.props.params
        return (
            <Modal
                visible={this.state.visible}
                destroyOnClose
                centered
                closable={false}
                width={1000}
                bodyStyle={{ width: '100%', height: '80vh' }}
                onCancel={() => { this.setState({ visible: false }) }}
                afterClose={this.props.onDestroy}
                footer={null}
            >
                <iframe
                    src={href}
                    title="previewer"
                    style={{ width: '100%', height: '100%' }}
                />
            </Modal>
        )
    }
}

export default PreviewPDFModal
