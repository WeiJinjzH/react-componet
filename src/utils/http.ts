import 'whatwg-fetch'
import utils from './utils'

class Http {
    defaultOptions = {
        credentials: 'include',
        method: 'GET',
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
        },
    }

    get(_url: string, urlParams?: {[key: string]: any}) {
        const url = utils.setUrlParams(_url, urlParams)
        return this.request(url)
    }

    post(_url: string, data?: {[key: string]: any}, urlParams?: {[key: string]: any}) {
        const url = utils.setUrlParams(_url, urlParams)
        const options = {
            method: 'POST',
            body: JSON.stringify(data),
        }
        return this.request(url, options)
    }

    delete(_url: string, urlParams?: {[key: string]: any}) {
        const url = utils.setUrlParams(_url, urlParams)
        const options = {
            method: 'DELETE',
        }
        return this.request(url, options)
    }

    put(_url, data?: {[key: string]: any}, params?: {[key: string]: any}) {
        const url = utils.setUrlParams(_url, params)
        const options = {
            method: 'PUT',
            body: JSON.stringify(data),
        }
        return this.request(url, options)
    }

    async request(_url: string, options = {}) {
        let url = _url
        if (_url.slice(0, 1) === '/') {
            url = utils.getBaseURL() + _url
        }
        let fetchOptions = {}
        // const regResult = document.cookie.match(/TOKEN_MANAGE=(.+?)(;|$)/)
        fetchOptions = Object.assign(fetchOptions, this.defaultOptions, options)
        // fetchOptions.headers.TOKEN = regResult && regResult[1]
        try {
            const res = await fetch(url, fetchOptions)
            if (res.status >= 200 && res.status < 300) {
                const contentType = res.headers.get('content-type') || ''
                if (contentType.indexOf('text/html') > -1) {
                    return res.text().then((text) => {
                        if (text.indexOf('请重新登录') > -1) {
                            utils.showSingleModal('error', {
                                title: '登录已失效，请重新登录',
                            })
                        }
                    })
                }
                return res.json().then((data) => data)
            }
            utils.showSingleModal('error', {
                title: `${res.status}`,
                content: res.statusText || '网络错误，请检查网络或接口',
                okText: '确定',
            })
            return Promise.reject(res)
        } catch (err) {
            utils.showSingleModal('error', {
                title: '网络错误，请检查网络或接口',
            })
            return Promise.reject(err)
        }
    }
}

const http = new Http()

export { http }

export default http
