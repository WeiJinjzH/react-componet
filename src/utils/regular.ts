export default {
    // 手机号码
    isMobile(value) {
        return (value.length === 11 && /^((13|18|15)\d{9})|(145|147|170|171|173|175|176|177|178)\d{8}$/.test(value))
    },
    validPassword: (value) => value.length >= 8 && value.length <= 16 && /^[\w_-]{8,16}$/.test(value),
    validStrictPassword: (value) => value.length >= 8 && value.length <= 16 && /(?![^a-zA-Z]+$)(?!\\D+$)^(?!.*\\s).{8,16}/.test(value),
    validateIDCard: (value) => value.length <= 18 && /^(^[1-9]\d{7}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])\d{3}$)|(^[1-9]\d{5}[1-9]\d{3}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])((\d{4})|\d{3}[X])$)$/.test(value),
    validateHongKongMacauPass: (value) => value.length <= 11 && /^[HMhm]{1}([0-9]{10}|[0-9]{8})$/.test(value),
}
