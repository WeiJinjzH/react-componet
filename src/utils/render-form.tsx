import React from 'react'
import { Form, Row, Col } from 'antd'
import { WrappedFormUtils } from 'antd/lib/form/Form'
import { ColProps } from 'antd/lib/grid/col'
import renderFormItem, { FormItemProps } from './render-form-item'

type ColSpanType = number | string;

export interface LayoutProps {
    labelCol?: { span: ColSpanType };
    wrapperCol?: { span: ColSpanType };
}

export interface FormProps {
    colSpan?: ColSpanType | ColProps;
    layout?: LayoutProps;
    isInline?: boolean;
    readonly?: boolean;
    labelAlign?: 'left' | 'right';
}

/**
 * 表单渲染工具
 */
function renderForm(form: WrappedFormUtils, formItems: Array<FormItemProps>, formProps: FormProps = {}) {
    let formColProps = formProps.colSpan
    if (typeof formColProps === 'number') {
        formColProps = {
            span: formColProps,
        }
    }
    return (
        <Form labelAlign={formProps.labelAlign || 'right'} className={`render-form${formProps.readonly ? ' render-form--readonly' : ''}`}>
            <Row type="flex" style={{ flexWrap: 'wrap' }}>
                {
                    formItems.map((item) => {
                        if (item.hidden) {
                            return null
                        }
                        if (item.type === 'WhiteSpace') {
                            return <div key={item.key} style={{ height: item.height || 0, width: '100%', clear: 'both' }} />
                        }
                        let itemColProps = item.colSpan
                        if (typeof itemColProps === 'number') {
                            itemColProps = {
                                span: item.colSpan,
                            }
                        }
                        const colProps = itemColProps || formColProps || { span: 12 }
                        return (
                            <Col
                                key={item.key}
                                {...colProps}
                            >
                                { renderFormItem(form, item, formProps) }
                            </Col>
                        )
                    })
                }
            </Row>
        </Form>
    )
}

export default renderForm
