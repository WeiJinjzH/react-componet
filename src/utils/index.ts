import utils from './utils'
import http from './http'
import * as validate from './validate'

export {
    http,
    validate,
}

export default utils
