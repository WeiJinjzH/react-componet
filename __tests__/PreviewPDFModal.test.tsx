import { render, shallow, mount } from 'enzyme'
import React from 'react'
import utils from 'src/utils'
import PreviewPDFModal from 'src/components/PreviewPDFModal'

describe('utils function: PreviewPDFModal', () => {
    it('PreviewPDFModal render test', () => {
        const onCancel = jest.fn()
        const wrapper = mount(<PreviewPDFModal href=".pdf" onCancel={onCancel} />)
        expect(wrapper.find('.ant-modal-wrap')).toHaveLength(1)
        expect(wrapper).toMatchSnapshot()
        wrapper.props().onCancel()
        expect(onCancel).toBeCalledTimes(1)
    })
})

describe('Component: PreviewPDFModal', () => {
    it('render test', () => {
        const wrapper1 = mount(<PreviewPDFModal visible href=".pdf" />)
        expect(wrapper1).toMatchSnapshot()

        const wrapper2 = mount(<PreviewPDFModal href=".pdf" />)
        expect(wrapper2).toMatchSnapshot()
    })
    it('interaction test: Modal onCancel event', () => {
        const onCancel = jest.fn()
        const onDestroy = jest.fn()
        const wrapper = mount(<PreviewPDFModal visible href=".jpg" onCancel={onCancel} onDestroy={onDestroy} />)
        expect(wrapper).toMatchSnapshot()
        wrapper.children().at(0).props().onCancel()
        expect(onCancel).toBeCalledTimes(1)
        wrapper.children().at(0).props().afterClose()
        expect(onDestroy).toBeCalledTimes(1)
    })
})
