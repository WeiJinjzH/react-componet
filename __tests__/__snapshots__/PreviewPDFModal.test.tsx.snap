// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PreviewPDFModal interaction test: Modal onCancel event 1`] = `
<PreviewPDFModal
  href=".jpg"
  onCancel={[MockFunction]}
  onDestroy={[MockFunction]}
  visible={true}
>
  <Modal
    afterClose={[MockFunction]}
    bodyStyle={
      Object {
        "height": "80vh",
        "width": "100%",
      }
    }
    centered={true}
    closable={false}
    confirmLoading={false}
    destroyOnClose={true}
    footer={null}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    transitionName="zoom"
    visible={true}
    width={1000}
  >
    <Component
      afterClose={[MockFunction]}
      bodyStyle={
        Object {
          "height": "80vh",
          "width": "100%",
        }
      }
      closable={false}
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Close
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      destroyOnClose={true}
      footer={null}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      transitionName="zoom"
      visible={true}
      width={1000}
      wrapClassName="ant-modal-centered"
    >
      <PortalWrapper
        visible={true}
      >
        <Portal
          getContainer={[Function]}
        >
          <Portal
            containerInfo={
              <div>
                <div>
                  <div
                    class="ant-modal-mask fade-appear"
                  />
                  <div
                    class="ant-modal-wrap ant-modal-centered"
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="ant-modal zoom-appear"
                      role="document"
                      style="width: 1000px;"
                    >
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                      <div
                        class="ant-modal-content"
                      >
                        <div
                          class="ant-modal-body"
                          style="width: 100%; height: 80vh;"
                        >
                          <iframe
                            src=".jpg"
                            style="width: 100%; height: 100%;"
                            title="previewer"
                          />
                        </div>
                      </div>
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <Dialog
              afterClose={[MockFunction]}
              bodyStyle={
                Object {
                  "height": "80vh",
                  "width": "100%",
                }
              }
              className=""
              closable={false}
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <Close
                    className="ant-modal-close-icon"
                  />
                </span>
              }
              confirmLoading={false}
              destroyOnClose={true}
              footer={null}
              getContainer={[Function]}
              getOpenCount={[Function]}
              keyboard={true}
              mask={true}
              maskClosable={true}
              maskTransitionName="fade"
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              prefixCls="ant-modal"
              transitionName="zoom"
              visible={true}
              width={1000}
              wrapClassName="ant-modal-centered"
            >
              <div>
                <Animate
                  animation={Object {}}
                  component=""
                  componentProps={Object {}}
                  key="mask"
                  onAppear={[Function]}
                  onEnd={[Function]}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  showProp="visible"
                  transitionAppear={true}
                  transitionEnter={true}
                  transitionLeave={true}
                  transitionName="fade"
                >
                  <AnimateChild
                    animation={Object {}}
                    key="mask"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="fade"
                  >
                    <LazyRenderBox
                      className="ant-modal-mask"
                      hiddenClassName="ant-modal-mask-hidden"
                      key="mask"
                      style={Object {}}
                      visible={true}
                    >
                      <div
                        className="ant-modal-mask"
                        style={Object {}}
                      />
                    </LazyRenderBox>
                  </AnimateChild>
                </Animate>
                <div
                  aria-labelledby={null}
                  className="ant-modal-wrap ant-modal-centered"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                  tabIndex={-1}
                >
                  <Animate
                    animation={Object {}}
                    component=""
                    componentProps={Object {}}
                    key="dialog"
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="visible"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="zoom"
                  >
                    <AnimateChild
                      animation={Object {}}
                      key="dialog-element"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="zoom"
                    >
                      <LazyRenderBox
                        className="ant-modal "
                        key="dialog-element"
                        onMouseDown={[Function]}
                        role="document"
                        style={
                          Object {
                            "width": 1000,
                          }
                        }
                        visible={true}
                      >
                        <div
                          className="ant-modal "
                          onMouseDown={[Function]}
                          role="document"
                          style={
                            Object {
                              "width": 1000,
                            }
                          }
                        >
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                          <div
                            className="ant-modal-content"
                          >
                            <div
                              className="ant-modal-body"
                              style={
                                Object {
                                  "height": "80vh",
                                  "width": "100%",
                                }
                              }
                            >
                              <iframe
                                src=".jpg"
                                style={
                                  Object {
                                    "height": "100%",
                                    "width": "100%",
                                  }
                                }
                                title="previewer"
                              />
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                        </div>
                      </LazyRenderBox>
                    </AnimateChild>
                  </Animate>
                </div>
              </div>
            </Dialog>
          </Portal>
        </Portal>
      </PortalWrapper>
    </Component>
  </Modal>
</PreviewPDFModal>
`;

exports[`Component: PreviewPDFModal render test 1`] = `
<PreviewPDFModal
  href=".pdf"
  visible={true}
>
  <Modal
    bodyStyle={
      Object {
        "height": "80vh",
        "width": "100%",
      }
    }
    centered={true}
    closable={false}
    confirmLoading={false}
    destroyOnClose={true}
    footer={null}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    transitionName="zoom"
    visible={true}
    width={1000}
  >
    <Component
      bodyStyle={
        Object {
          "height": "80vh",
          "width": "100%",
        }
      }
      closable={false}
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Close
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      destroyOnClose={true}
      footer={null}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      transitionName="zoom"
      visible={true}
      width={1000}
      wrapClassName="ant-modal-centered"
    >
      <PortalWrapper
        visible={true}
      >
        <Portal
          getContainer={[Function]}
        >
          <Portal
            containerInfo={
              <div>
                <div>
                  <div
                    class="ant-modal-mask fade-appear"
                  />
                  <div
                    class="ant-modal-wrap ant-modal-centered"
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="ant-modal zoom-appear"
                      role="document"
                      style="width: 1000px;"
                    >
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                      <div
                        class="ant-modal-content"
                      >
                        <div
                          class="ant-modal-body"
                          style="width: 100%; height: 80vh;"
                        >
                          <iframe
                            src=".pdf"
                            style="width: 100%; height: 100%;"
                            title="previewer"
                          />
                        </div>
                      </div>
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <Dialog
              bodyStyle={
                Object {
                  "height": "80vh",
                  "width": "100%",
                }
              }
              className=""
              closable={false}
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <Close
                    className="ant-modal-close-icon"
                  />
                </span>
              }
              confirmLoading={false}
              destroyOnClose={true}
              footer={null}
              getContainer={[Function]}
              getOpenCount={[Function]}
              keyboard={true}
              mask={true}
              maskClosable={true}
              maskTransitionName="fade"
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              prefixCls="ant-modal"
              transitionName="zoom"
              visible={true}
              width={1000}
              wrapClassName="ant-modal-centered"
            >
              <div>
                <Animate
                  animation={Object {}}
                  component=""
                  componentProps={Object {}}
                  key="mask"
                  onAppear={[Function]}
                  onEnd={[Function]}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  showProp="visible"
                  transitionAppear={true}
                  transitionEnter={true}
                  transitionLeave={true}
                  transitionName="fade"
                >
                  <AnimateChild
                    animation={Object {}}
                    key="mask"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="fade"
                  >
                    <LazyRenderBox
                      className="ant-modal-mask"
                      hiddenClassName="ant-modal-mask-hidden"
                      key="mask"
                      style={Object {}}
                      visible={true}
                    >
                      <div
                        className="ant-modal-mask"
                        style={Object {}}
                      />
                    </LazyRenderBox>
                  </AnimateChild>
                </Animate>
                <div
                  aria-labelledby={null}
                  className="ant-modal-wrap ant-modal-centered"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                  tabIndex={-1}
                >
                  <Animate
                    animation={Object {}}
                    component=""
                    componentProps={Object {}}
                    key="dialog"
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="visible"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="zoom"
                  >
                    <AnimateChild
                      animation={Object {}}
                      key="dialog-element"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="zoom"
                    >
                      <LazyRenderBox
                        className="ant-modal "
                        key="dialog-element"
                        onMouseDown={[Function]}
                        role="document"
                        style={
                          Object {
                            "width": 1000,
                          }
                        }
                        visible={true}
                      >
                        <div
                          className="ant-modal "
                          onMouseDown={[Function]}
                          role="document"
                          style={
                            Object {
                              "width": 1000,
                            }
                          }
                        >
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                          <div
                            className="ant-modal-content"
                          >
                            <div
                              className="ant-modal-body"
                              style={
                                Object {
                                  "height": "80vh",
                                  "width": "100%",
                                }
                              }
                            >
                              <iframe
                                src=".pdf"
                                style={
                                  Object {
                                    "height": "100%",
                                    "width": "100%",
                                  }
                                }
                                title="previewer"
                              />
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                        </div>
                      </LazyRenderBox>
                    </AnimateChild>
                  </Animate>
                </div>
              </div>
            </Dialog>
          </Portal>
        </Portal>
      </PortalWrapper>
    </Component>
  </Modal>
</PreviewPDFModal>
`;

exports[`Component: PreviewPDFModal render test 2`] = `
<PreviewPDFModal
  href=".pdf"
>
  <Modal
    bodyStyle={
      Object {
        "height": "80vh",
        "width": "100%",
      }
    }
    centered={true}
    closable={false}
    confirmLoading={false}
    destroyOnClose={true}
    footer={null}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    transitionName="zoom"
    visible={true}
    width={1000}
  >
    <Component
      bodyStyle={
        Object {
          "height": "80vh",
          "width": "100%",
        }
      }
      closable={false}
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Close
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      destroyOnClose={true}
      footer={null}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      transitionName="zoom"
      visible={true}
      width={1000}
      wrapClassName="ant-modal-centered"
    >
      <PortalWrapper
        visible={true}
      >
        <Portal
          getContainer={[Function]}
        >
          <Portal
            containerInfo={
              <div>
                <div>
                  <div
                    class="ant-modal-mask fade-appear"
                  />
                  <div
                    class="ant-modal-wrap ant-modal-centered"
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="ant-modal zoom-appear"
                      role="document"
                      style="width: 1000px;"
                    >
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                      <div
                        class="ant-modal-content"
                      >
                        <div
                          class="ant-modal-body"
                          style="width: 100%; height: 80vh;"
                        >
                          <iframe
                            src=".pdf"
                            style="width: 100%; height: 100%;"
                            title="previewer"
                          />
                        </div>
                      </div>
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <Dialog
              bodyStyle={
                Object {
                  "height": "80vh",
                  "width": "100%",
                }
              }
              className=""
              closable={false}
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <Close
                    className="ant-modal-close-icon"
                  />
                </span>
              }
              confirmLoading={false}
              destroyOnClose={true}
              footer={null}
              getContainer={[Function]}
              getOpenCount={[Function]}
              keyboard={true}
              mask={true}
              maskClosable={true}
              maskTransitionName="fade"
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              prefixCls="ant-modal"
              transitionName="zoom"
              visible={true}
              width={1000}
              wrapClassName="ant-modal-centered"
            >
              <div>
                <Animate
                  animation={Object {}}
                  component=""
                  componentProps={Object {}}
                  key="mask"
                  onAppear={[Function]}
                  onEnd={[Function]}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  showProp="visible"
                  transitionAppear={true}
                  transitionEnter={true}
                  transitionLeave={true}
                  transitionName="fade"
                >
                  <AnimateChild
                    animation={Object {}}
                    key="mask"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="fade"
                  >
                    <LazyRenderBox
                      className="ant-modal-mask"
                      hiddenClassName="ant-modal-mask-hidden"
                      key="mask"
                      style={Object {}}
                      visible={true}
                    >
                      <div
                        className="ant-modal-mask"
                        style={Object {}}
                      />
                    </LazyRenderBox>
                  </AnimateChild>
                </Animate>
                <div
                  aria-labelledby={null}
                  className="ant-modal-wrap ant-modal-centered"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                  tabIndex={-1}
                >
                  <Animate
                    animation={Object {}}
                    component=""
                    componentProps={Object {}}
                    key="dialog"
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="visible"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="zoom"
                  >
                    <AnimateChild
                      animation={Object {}}
                      key="dialog-element"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="zoom"
                    >
                      <LazyRenderBox
                        className="ant-modal "
                        key="dialog-element"
                        onMouseDown={[Function]}
                        role="document"
                        style={
                          Object {
                            "width": 1000,
                          }
                        }
                        visible={true}
                      >
                        <div
                          className="ant-modal "
                          onMouseDown={[Function]}
                          role="document"
                          style={
                            Object {
                              "width": 1000,
                            }
                          }
                        >
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                          <div
                            className="ant-modal-content"
                          >
                            <div
                              className="ant-modal-body"
                              style={
                                Object {
                                  "height": "80vh",
                                  "width": "100%",
                                }
                              }
                            >
                              <iframe
                                src=".pdf"
                                style={
                                  Object {
                                    "height": "100%",
                                    "width": "100%",
                                  }
                                }
                                title="previewer"
                              />
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                        </div>
                      </LazyRenderBox>
                    </AnimateChild>
                  </Animate>
                </div>
              </div>
            </Dialog>
          </Portal>
        </Portal>
      </PortalWrapper>
    </Component>
  </Modal>
</PreviewPDFModal>
`;

exports[`utils function: PreviewPDFModal PreviewPDFModal render test 1`] = `
<PreviewPDFModal
  href=".pdf"
  onCancel={[MockFunction]}
>
  <Modal
    bodyStyle={
      Object {
        "height": "80vh",
        "width": "100%",
      }
    }
    centered={true}
    closable={false}
    confirmLoading={false}
    destroyOnClose={true}
    footer={null}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    transitionName="zoom"
    visible={true}
    width={1000}
  >
    <Component
      bodyStyle={
        Object {
          "height": "80vh",
          "width": "100%",
        }
      }
      closable={false}
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Close
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      destroyOnClose={true}
      footer={null}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      transitionName="zoom"
      visible={true}
      width={1000}
      wrapClassName="ant-modal-centered"
    >
      <PortalWrapper
        visible={true}
      >
        <Portal
          getContainer={[Function]}
        >
          <Portal
            containerInfo={
              <div>
                <div>
                  <div
                    class="ant-modal-mask fade-appear"
                  />
                  <div
                    class="ant-modal-wrap ant-modal-centered"
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="ant-modal zoom-appear"
                      role="document"
                      style="width: 1000px;"
                    >
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                      <div
                        class="ant-modal-content"
                      >
                        <div
                          class="ant-modal-body"
                          style="width: 100%; height: 80vh;"
                        >
                          <iframe
                            src=".pdf"
                            style="width: 100%; height: 100%;"
                            title="previewer"
                          />
                        </div>
                      </div>
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <Dialog
              bodyStyle={
                Object {
                  "height": "80vh",
                  "width": "100%",
                }
              }
              className=""
              closable={false}
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <Close
                    className="ant-modal-close-icon"
                  />
                </span>
              }
              confirmLoading={false}
              destroyOnClose={true}
              footer={null}
              getContainer={[Function]}
              getOpenCount={[Function]}
              keyboard={true}
              mask={true}
              maskClosable={true}
              maskTransitionName="fade"
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              prefixCls="ant-modal"
              transitionName="zoom"
              visible={true}
              width={1000}
              wrapClassName="ant-modal-centered"
            >
              <div>
                <Animate
                  animation={Object {}}
                  component=""
                  componentProps={Object {}}
                  key="mask"
                  onAppear={[Function]}
                  onEnd={[Function]}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  showProp="visible"
                  transitionAppear={true}
                  transitionEnter={true}
                  transitionLeave={true}
                  transitionName="fade"
                >
                  <AnimateChild
                    animation={Object {}}
                    key="mask"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="fade"
                  >
                    <LazyRenderBox
                      className="ant-modal-mask"
                      hiddenClassName="ant-modal-mask-hidden"
                      key="mask"
                      style={Object {}}
                      visible={true}
                    >
                      <div
                        className="ant-modal-mask"
                        style={Object {}}
                      />
                    </LazyRenderBox>
                  </AnimateChild>
                </Animate>
                <div
                  aria-labelledby={null}
                  className="ant-modal-wrap ant-modal-centered"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                  tabIndex={-1}
                >
                  <Animate
                    animation={Object {}}
                    component=""
                    componentProps={Object {}}
                    key="dialog"
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="visible"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="zoom"
                  >
                    <AnimateChild
                      animation={Object {}}
                      key="dialog-element"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="zoom"
                    >
                      <LazyRenderBox
                        className="ant-modal "
                        key="dialog-element"
                        onMouseDown={[Function]}
                        role="document"
                        style={
                          Object {
                            "width": 1000,
                          }
                        }
                        visible={true}
                      >
                        <div
                          className="ant-modal "
                          onMouseDown={[Function]}
                          role="document"
                          style={
                            Object {
                              "width": 1000,
                            }
                          }
                        >
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                          <div
                            className="ant-modal-content"
                          >
                            <div
                              className="ant-modal-body"
                              style={
                                Object {
                                  "height": "80vh",
                                  "width": "100%",
                                }
                              }
                            >
                              <iframe
                                src=".pdf"
                                style={
                                  Object {
                                    "height": "100%",
                                    "width": "100%",
                                  }
                                }
                                title="previewer"
                              />
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                        </div>
                      </LazyRenderBox>
                    </AnimateChild>
                  </Animate>
                </div>
              </div>
            </Dialog>
          </Portal>
        </Portal>
      </PortalWrapper>
    </Component>
  </Modal>
</PreviewPDFModal>
`;
